Controller/Payment/Webhook.php

```diff
@@ -237,8 +238,11 @@ protected function authorize(array $post)
                 {
                     $payment = $order->getPayment();
 
-                    if (($order->getStatus() === static::STATUS_PENDING) and
-                        (empty($payment->getLastTransId()) === true) )
+                    if ((($order->getState() === static::STATE_NEW
+                        && $order->getStatus() === static::STATUS_CANCELED)
```

Comment:
@enishant please use `and` and `or`

Reply:
@ramth05  updated condition and fixed formatting issue.